### ISSUE TITLE
Dependency issues on new install with ivar conda environment

### ISSUE THREAD

**Issue by dgaston** [jaleezyy/covid-19-signal#140]
After running into some issues on an update with Nextclade, I wanted to start the pipeline fresh. Re-creating the signal conda environment and a fresh github checkout of the repository. When running ./signalexe.py install I am running into the following issue when it is setting up the ivar conda environment:

LibMambaUnsatisfiableError: Encountered problems while solving:
  - package ivar-1.3.1-ha04fe3b_4 requires htslib >=1.16,<1.20.0a0, but none of the providers can be installed

Could not solve for environment specs
The following packages are incompatible
└─ ivar 1.3**  is not installable because there are no viable options
   ├─ ivar [1.3|1.3.1] would require
   │  └─ libdeflate >=1.6,<1.7.0a0 , which conflicts with any installable versions previously reported;
   ├─ ivar 1.3.1 would require
   │  └─ htslib [>=1.12,<1.20.0a0 |>=1.14,<1.20.0a0 ] but there are no viable options
   │     ├─ htslib [1.15.1|1.16|1.17] would require
   │     │  └─ libdeflate >=1.13,<1.20.0a0 , which conflicts with any installable versions previously reported;
   │     ├─ htslib [1.17|1.18|1.19] would require
   │     │  └─ libdeflate >=1.18,<1.20.0a0 , which conflicts with any installable versions previously reported;
   │     ├─ htslib 1.19.1 would require
   │     │  └─ libdeflate >=1.18,<1.19.0a0 , which conflicts with any installable versions previously reported;
   │     ├─ htslib [1.12|1.13|1.14] would require
   │     │  └─ libdeflate >=1.7,<1.20.0a0 , which conflicts with any installable versions previously reported;
   │     ├─ htslib 1.14 would require
   │     │  └─ libdeflate >=1.9,<1.20.0a0 , which conflicts with any installable versions previously reported;
   │     └─ htslib [1.14|1.15|1.15.1] would require
   │        └─ libdeflate >=1.10,<1.20.0a0 , which conflicts with any installable versions previously reported;
   └─ ivar [1.3.1|1.3.2] would require
      └─ htslib >=1.16,<1.20.0a0 , which cannot be installed (as previously explained).

Has anyone tested with newer versions of iVar, htslib, etc in these environments?

Also I am curious what may be the root cause here. It isn't like these environment.yaml files are updating particularly frequently.

**Comment by jaleezyy** [comment]
Interesting outcome given you'd think that older versions and corresponding compatibilities would be maintained but sometimes version specifications can change, something we both can't always anticipate. Noticing this error is from mamba and perhaps may differ versus conda in terms of different conditions for satisfying dependencies. These are my initial thoughts looking at your error. 

**I'll try and recreate your error and see if some tweaks are needed in the ivar dependencies.** 

In the meanwhile, perhaps by removing the `ivar=1.3` to just `ivar` under the [`conda_envs/ivar.yaml`](https://github.com/jaleezyy/covid-19-signal/blob/master/conda_envs/ivar.yaml) environment yaml file may overcome this error and let mamba/conda find the optimal version setup.

**Comment by dgaston** [comment]
Changing the ivar version requirements doesn't change things, it is the samtools version number that seems to be where the conflict crops up with the chain through htslib and libdeflate. Removing the samtools version number requirement solves the problem. Fingers crossed that doesn't break anything else

**Comment by jaleezyy** [comment]
For a general assessment: could you list (`conda list` or `mamba list`) the environment you could get running? Could you also clarify the channel priority used (`conda config --show channel_priority`) if conda is involved? 

Admittingly, I haven't been able to recreate the UnsatisfiableError via conda or mamba. Looking at the software versions that do install:

```
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       2_gnu    conda-forge
bzip2                     1.0.8                hd590300_5    conda-forge
c-ares                    1.27.0               hd590300_0    conda-forge
ca-certificates           2024.2.2             hbcca054_0    conda-forge
htslib                    1.19.1               h81da01d_2    bioconda
ivar                      1.3.2                ha04fe3b_1    bioconda
keyutils                  1.6.1                h166bdaf_0    conda-forge
krb5                      1.21.2               h659d440_0    conda-forge
libcurl                   8.5.0                hca28451_0    conda-forge
libdeflate                1.18                 h0b41bf4_0    conda-forge
libedit                   3.1.20191231         he28a2e2_2    conda-forge
libev                     4.33                 hd590300_2    conda-forge
libgcc-ng                 13.2.0               h807b86a_5    conda-forge
libgomp                   13.2.0               h807b86a_5    conda-forge
libnghttp2                1.58.0               h47da74e_1    conda-forge
libssh2                   1.11.0               h0841786_0    conda-forge
libstdcxx-ng              13.2.0               h7e041cc_5    conda-forge
libzlib                   1.2.13               hd590300_5    conda-forge
ncurses                   6.2                  h58526e2_4    conda-forge
openssl                   3.2.1                hd590300_0    conda-forge
samtools                  1.10                 h2e538c0_3    bioconda
xz                        5.2.6                h166bdaf_0    conda-forge
zlib                      1.2.13               hd590300_5    conda-forge
zstd                      1.5.5                hfc55251_0    conda-forge
```

`ivar 1.3.2` seems to be the version taken along with `htslib 1.19.1` and `libdeflate 1.18`. 

Could you try setting the version to what I have noted above and seeing if that runs? 

If samtools is the problematic software and being hyper-specific isn't working, perhaps set `ivar=1.3` and set `samtools` with no specified version and perhaps we can get a version of this running?

**Comment by dgaston** [comment]
It looks like hyper specifying the version numbers worked. This appears to likely be an issue with mamba and channel priorities perhaps. I did have them set on strict, but that config no longer seems to be "sticking" and keeps reverting back to flexible. So that could be it too.

Thanks for your help! Seems to be a bit of a baffling random error, as I recently set up the pipeline elsewhere using mamba as well and everything worked fine.
